psst-core/src/player/queue.rs

```diff
@@ -51,6 +51,13 @@ impl Queue {
     }
 
     fn compute_positions(&mut self) {
+        // In the case of switching away from shuffle, the position should be set back to
+        // where it appears in the actual playlist order.
+        let playlist_position = if self.positions.len() > 1 {
+            self.positions[self.position]
+        } else {
+            self.position
+        };
```

Comment:
`self.position` can be set to the result of this directly since this let isn't used anywhere else.

Reply:
Actually, this wouldn't work, as it depends on the randomized order of `positions`, and that gets reset on line 62.